nengo/builder/neurons.py

```diff
@@ -133,9 +142,54 @@ def build_neurons(model, neurontype, neurons):
                 "currently supported." % (key, type(init).__name__)
             )
 
-    model.sig[neurons]["out"] = (
-        state["spikes"] if neurontype.spiking else state["rates"]
-    )
     model.add_op(
-        SimNeurons(neurons=neurontype, J=model.sig[neurons]["in"], state=state)
+        SimNeurons(
+            neurons=neurontype,
+            J=model.sig[neurons]["in"],
+            output=model.sig[neurons]["out"],
+            state=state,
+        )
+    )
+
+
+@Builder.register(RatesToSpikesNeuronType)
+def build_rates_to_spikes(model, neurontype, neurons):
+    """Builds a `.RatesToSpikesNeuronType` object into a model.
+
+    This function adds two `.SimNeurons` operators. The first one handles
+    simulating the base_type, converting input signals into rates. The second
+    one takes those rates as input and emits spikes.
+
+    Parameters
+    ----------
+    model : Model
+        The model to build into.
+    neurontype : RatesToSpikesNeuronType
+        Neuron type to build.
+    neuron : Neurons
+        The neuron population object corresponding to the neuron type.
+
+    Notes
+    -----
+    Does not modify ``model.params[]`` and can therefore be called
+    more than once with the same `.NeuronType` instance.
+    """
+
+    in_sig = model.sig[neurons]["in"]
+    out_sig = model.sig[neurons]["out"]
+    rate_sig = Signal(
+        shape=model.sig[neurons]["in"].shape, name="%s.rate_out" % (neurons,),
     )
+    model.sig[neurons]["rate_out"] = rate_sig
+
+    # build the base neuron type
+    # temporarily swap output signal so that base builder will output into rate_sig
+    model.sig[neurons]["out"] = rate_sig
```

Comment:
It would be cleaner IMO to modify `build_neurons` to take an optional `output` argument that defaults to `model.sig[neurons]["out"]` but can be set to `rate_sig` here.

Reply:
Thanks @hunse, fixup commit looks great!